File: course_1/week1/cpp/t4_lowest_common_multiple.cpp
/*
 least common multiple = abs(a * b)/gcd(a,b)
 
 */

#include <iostream>
#include <cmath> // for abs()
using namespace std;

long long euclid_GCD(long long num1, long long num2){
  if (num2 == 0){
    return num1;
  }
  
  long long num1_compliment;
  num1_compliment = num1 % num2;
  return euclid_GCD(num2, num1_compliment);
}

int main(){
  // gcd - greatest common divisor
  // lcm - least common multiple
  long long a, b, gcd, lcm;
  //cout << "Enter the first number: " << endl;
  cin >> a;
  //cout << "Enter the second number: " << endl;
  cin >> b;
  
  gcd = euclid_GCD(a, b);
//  cout << "The GCD of " << a << " and " << b << " is " << //result << endl;
  lcm = (long long)abs(a * b)/gcd;
  cout << lcm << endl;
  
  return 0;
  
  
}

File: course_1/week1/python/t4_lowest_common_multiple.py
# This implementation passed the grader
# least common multiple = abs(a * b)/gcd(a,b)

def euclid_GCD(num1, num2):
"""
Given two numbers calculates greatest common divisor
"""
  if num2 == 0:
    return num1
  
  num1_compliment = num1 % num2
  return euclid_GCD(num2, num1_compliment)

nums_input = input().split()
a = int(nums_input[0]) # input number 1
b = int(nums_input[1]) # input number 2
gcd = euclid_GCD(a,b) #greatest common divisor for a, b

lcm = int(abs(a * b)/ gcd) #lowest common multiple
print(lcm)
